document/FileFormat.mdk

```diff
@@ -2696,31 +2703,318 @@ The tool might provide a "related location" (&sect;[#result-relatedLocations]) w
+|                  |                             | of the value or values      | of any variables            | 
+|                  |                             | returned.                   | participating in the        |
+|                  |                             | Absent: If the function     | return expression           |
+|                  |                             | does not return a value.    | or expressions.             |
+|                  |                             |                             | Absent: If the function     |
+|                  |                             |                             | does not return a value.    |
+|------------------|-----------------------------|-----------------------------|-----------------------------|
+| `usage`          | Recommended: The            | Optional: A string          | Absent                      |
+|                  | name of the variable        | representation of the       |                             |
+|------------------|-----------------------------|-----------------------------|-----------------------------|
+
+~
+
+~IsoNumberedExample { @IsoNumberedExample=1 }
+In C++, if the source code contains the declaration
+
+```
+std::string &str = name;
+```
+
+then the value of `kind` would be `"alias"`, the value of `target` would be `"str"`, the value of `parameters` would be
+
+```
+[ "name" ]
+```
+
+and the value of `values` might be
+
+```
+{ "name": "\"John\"" }
+```
+~
+
+~IsoNumberedExample
+In C++, if the source code contains the declaration
+
+```
+std::string &str = name, &str2 = address;
+```
+
+and if the tool creating the log wished to represent both aliases in the log file, then the tool would create
+two `annotatedCodeLocation` objects, each with `kind` set to `"alias"`, and referring to the same source line.
+~
+
+~IsoNumberedExample
+In C++ or C#, if the source code contains the assignment
+
+```
+m = n + p;
+```
+
+then the value of `kind` would be `"assignment"`, the value of `target` would be `"m"`, the value of `parameters` might be
+
+```
+[ "5" ]
+```
+
+and the value of `variables` might be
+
+```
+{  "n": "2", "p": "3" }
+```
+~
+
+~IsoNumberedExample
+In C#, if the source code contains the test
+
+```
+if (s.Length > 0 && y > 2 && valid())
+```
+
+then the value of `kind` would be `"branch"`, `target` would be absent, the value of `parameters`
+might be
+
+```
+[ "true" ]
+```
+
+and the value of `variables` might be
+
+```
+{ "s": "\"A string\"", "y": "3" }
+```
+~
+
+~IsoNumberedExample
+In C++ or C#, if the source code contains the function call
+
+```
+f(7, m + n, "s", this, g(2));
+```
+
+then the value of `kind` would be `"call"`, the value of `target` might be `"f"` (or, for example, `"N.C.f"` if
+the function `f` occurred in class `C` in namespace `N`), the value of `parameters` would be
+
+```
+[ "7", "m + n", "\"s\"", "this", "g(2)" ]
+```
+
+and the value of `variables` might be
+
+```
+{ "m": "2", "n": "3" }
```

Comment:
ah, see in this example, you are assuming that variables fills out distinct variable values. that could very well be true. but it would be fine to have this as well, i think. do you agree? 

{"m + n" : "5" }